migrations/migration.py

```diff
@@ -0,0 +1,139 @@
+import os
+from typing import List
+from datetime import datetime
+
+from isacc_messaging.audit import audit_entry
+from isacc_messaging.models.fhir import (
+    HAPI_request,
+    first_in_bundle,
+    next_in_bundle,
+    resolve_reference,
+)
+from migrations.migration_resource import MigrationManager
+
+class Migration:
+    def __init__(self):
+        self.migrations_dir = os.path.join(os.path.dirname(__file__), "versions")
+        self.latest_applied_migration = self.get_applied_migration_from_fhir()
+
+    def get_migration_files(self) -> List[str]:
+        """Retrieve the list of migration files."""
+        migration_files = sorted(os.listdir(self.migrations_dir))
```

Comment:
great progress, i like where this is going.

i believe you'll need to look for the `revision` and `down_revision` numbers in the files to get the order, confirm no cycles, etc.

Reply:
probably worth adding a few mock test files, to confirm your `Migration.migration_files` end up ordered respecting the `down_revision` reverse order.  for bonus points, confirm it errors if there's a gap or cycle.